core/Azure.Mcp.Core/tests/Azure.Mcp.Core.UnitTests/Infrastructure/CommandMetadataSyncTests.cs

```diff
@@ -0,0 +1,120 @@
+{
+    private static readonly string _repoRoot = GetRepoRoot();
+
+    [Fact]
+    public void AzCommandsMetadata_Should_Be_Synchronized()
+    {
+        // Arrange
+        var serverProjectPath = Path.Combine(_repoRoot, "servers", "Azure.Mcp.Server", "src", "Azure.Mcp.Server.csproj");
+        var docsPath = Path.Combine(_repoRoot, "servers", "Azure.Mcp.Server", "docs", "azmcp-commands.md");
+        var updateScriptPath = Path.Combine(_repoRoot, "eng", "scripts", "Update-AzCommandsMetadata.ps1");
+
+        // Verify files exist
+        Assert.True(File.Exists(serverProjectPath), $"Server project not found at {serverProjectPath}");
+        Assert.True(File.Exists(docsPath), $"Documentation file not found at {docsPath}");
+        Assert.True(File.Exists(updateScriptPath), $"Update script not found at {updateScriptPath}");
+
+        // Act - Build the server project
+        var buildResult = RunCommand("dotnet", $"build \"{serverProjectPath}\" -c Debug", _repoRoot);
+
+        Assert.True(buildResult.ExitCode == 0, $"Build failed with exit code {buildResult.ExitCode}. Output: {buildResult.Output}. Error: {buildResult.Error}");
+
+        // Determine the executable path (OS-specific)
+        var exeName = OperatingSystem.IsWindows() ? "azmcp.exe" : "azmcp";
+        var azmcpPath = Path.Combine(_repoRoot, "servers", "Azure.Mcp.Server", "src", "bin", "Debug", "net9.0", exeName);
+
+        Assert.True(File.Exists(azmcpPath), $"Executable not found at {azmcpPath} after build");
+
+        // Get the original content before running the update script
+        var originalContent = File.ReadAllText(docsPath);
+
+        // Act - Run the update script
+        var updateResult = RunCommand(
+            "pwsh",
+            $"-NoProfile -ExecutionPolicy Bypass -File \"{updateScriptPath}\" -AzmcpPath \"{azmcpPath}\" -DocsPath \"{docsPath}\"",
+            _repoRoot);
+
+        Assert.True(updateResult.ExitCode == 0,
+            $"Update script failed with exit code {updateResult.ExitCode}. Output: {updateResult.Output}. Error: {updateResult.Error}");
+
+        // Assert - Check if the file was modified
+        var updatedContent = File.ReadAllText(docsPath);
+
+        Assert.True(originalContent == updatedContent,
+            "The azmcp-commands.md file is out of sync with tool metadata.\n\n" +
+            "To fix this issue:\n" +
+            "  1. Run the following command from the repository root:\n" +
+            $"     .\\eng\\scripts\\Update-AzCommandsMetadata.ps1\n" +
+            "  2. Review the changes to servers\\Azure.Mcp.Server\\docs\\azmcp-commands.md\n" +
+            "  3. Commit the updated file with your changes\n\n" +
+            "This ensures that tool metadata in the documentation stays synchronized with the actual command implementations.");
+    }
+
+    private static (int ExitCode, string Output, string Error) RunCommand(string command, string arguments, string workingDirectory)
```

Comment:
nit: I wonder if we should have a common helper for this? Or if we can use https://github.com/microsoft/mcp/blob/main/core/Azure.Mcp.Core/src/Services/ProcessExecution/ExternalProcessService.cs.